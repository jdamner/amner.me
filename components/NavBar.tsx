import React from "react";
/* Components */
import { HomeIcon } from "@heroicons/react/24/solid";
import InlineLink from "./Atoms/Links/Inline";

export default function NavBar(
  props: React.DetailedHTMLProps<
    React.HTMLAttributes<HTMLElement>,
    HTMLElement
  >,
) {
  const { children, ...rest } = props;
  return (
    <nav
      className="w-full h-16 bg-gradient-to-r text-slate-100 from-slate-900 to-slate-700 dark:text-slate-200"
      id="main-nav"
      {...rest}
    >
      <div className="container mx-auto h-full flex flex-row items-center px-3 md:px-0">
        <div className="">
          <InlineLink href="/" className="text-slate-100 hover:text-slate-300">
            <HomeIcon className="h-10" />
            <span className="sr-only">Home</span>
          </InlineLink>
        </div>
        <div className="ml-auto">{children}</div>
      </div>
    </nav>
  );
}
