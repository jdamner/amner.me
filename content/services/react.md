---
title: Javascript & React
---

I've worked with [React](https://reactjs.org) and [React Native](https://reactnative.dev) for a number of years and find the stateful nature of the React framework to really work well in my head. I love the component based approach and the ability to create reusable components that can be used in multiple places.

This site is build using React and the [NextJS](https://www.nextjs.org/) framework. I parse [markdown](https://daringfireball.net/projects/markdown/) files for the content. This site is totally open source too, so if you're curious about how it's built, you can check out the [source code](https://www.github.com/jdamner/amner.me).

Also, as a statically generated site, it's super fast and easy to deploy.
