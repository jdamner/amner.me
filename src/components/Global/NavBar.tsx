import Link from "next/link"
import InlineLink from "../Links/Inline";

export default function NavBar(props): JSX.Element {
	return (
		<nav className="w-full h-16 bg-gradient-to-r bg-emerald-900 text-slate-100 from-teal-900 to-emerald-600">
			<div className='container mx-auto h-full flex flex-row justify-between items-center px-3 md:px-0'>
				<ul>
					<li className="inline-block mr-3">
						<InlineLink href="/" className="text-xl font-bold">amner.me</InlineLink>
					</li>
				</ul>
				{props.children}
			</div>
		</nav>
	);
}
