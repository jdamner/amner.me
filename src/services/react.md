---
title: React
thumbnail: ../assets/react-logo.png
---

Working with SQL in raw queries, and using frameworks such as Eloquent and WP_Query.
Coaching and supporting a team of developers grow their skills and meet their objectives
Supporting Project Managers and Business Development teams for technical guidance on quoting and supporting project lifecycles.
Debugging and maintaining PHP projects built in many frameworks, including bespoke PHP frameworks developed in house.
Developing systems that interact with complex data, such as data provided via third-party tools and services that run on device and output in raw formats.