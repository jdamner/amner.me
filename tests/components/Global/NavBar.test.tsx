import { render } from "@testing-library/react"
import NavBar from "../../../src/components/Global/NavBar"

describe("NavBar", () => {
  it("renders correctly", () => {
    const { container } = render(<NavBar><a href="/">Home</a></NavBar>);
    expect(container).toMatchSnapshot()
  })
})
