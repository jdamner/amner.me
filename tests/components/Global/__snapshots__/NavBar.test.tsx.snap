// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly 1`] = `
<div>
  <nav
    class="w-full h-16 bg-gradient-to-r bg-emerald-900 text-slate-100 from-teal-900 to-emerald-600"
  >
    <div
      class="container mx-auto h-full flex flex-row justify-between items-center px-3 md:px-0"
    >
      <ul>
        <li
          class="inline-block mr-3"
        >
          <a
            class="text-xl font-bold"
            href="/"
          >
            amner.me
          </a>
        </li>
      </ul>
      <a
        href="/"
      >
        Home
      </a>
    </div>
  </nav>
</div>
`;
